cmake/gen_flasher_stub.py

```diff
@@ -0,0 +1,148 @@
+import base64
+import json
+import os
+import sys
+import urllib.request
+
+"""
+Generate flasher stubs from https://github.com/espressif/esptool/tree/master/esptool/targets/stub_flasher
+"""
+
+# Paths
+esptool_version = sys.argv[1]
+root_path = sys.argv[2]
+priv_inc_path = os.path.join(root_path, "private_include")
+src_path = os.path.join(root_path, "src")
+hfile_path = os.path.join(priv_inc_path, "esp_stubs.h")
+cfile_path = os.path.join(src_path, "esp_stubs.c")
+
+# Matches order of target_chip_t enumeration
+files_to_download = [
+    "stub_flasher_8266.json",  # ESP8266_CHIP
+    "stub_flasher_32.json",  # ESP32_CHIP
+    "stub_flasher_32s2.json",  # ESP32S2_CHIP
+    "stub_flasher_32c3.json",  # ESP32C3_CHIP
+    "stub_flasher_32s3.json",  # ESP32S3_CHIP
+    "stub_flasher_32c2.json",  # ESP32C2_CHIP
+    "reserved.json",  # ESP32_RESERVED0_CHIP
```

Comment:
The target which creates the stubs does not run during the build. I've intentionally chosen not to do that. If you don't explicitly invoke the "gen_flasher_stubs" target the stubs will not change! 

I'd have preferred to scan the GitHub repo and only download existing .json files, but I wanted to match the existing target enumeration order. That's also the reason why there absolutely needs to be a hole for the currently reserved target. Otherwise the following enum value would be off by one.

I agree that this isn't ideal though, but getting rid of the "reserved" value is not an option.

Reply:
For some reason my brain hallucinated the `PRE_BUILD` argument, my bad. 

Still, we should not fail by writing zeroes, but instead notify the user. We don't have to get rid of the "reserved" value, just add a check for it and work around it by writing `" // " + file_to_download + "\n" " {},\n" "\n"`.

Reply:
Codegen of the stubs now fails if the HTTP request fails for whatever reason.

https://github.com/higaski/esp-serial-flasher/commit/e4cbcae6c606b79850abc46dbd9a30457a613752